tasks/upload.py

```diff
@@ -540,95 +514,87 @@ def run_impl_within_lock(
 
         if argument_list:
             db_session.commit()
-            self.schedule_task(
+            scheduled_tasks = self.schedule_task(
+                db_session,
                 commit,
                 commit_yaml,
                 argument_list,
                 commit_report,
                 upload_context,
-                db_session,
                 checkpoints,
             )
+
+            log.info(
+                f"Scheduling {upload_context.report_type.value} processing tasks",
+                extra=upload_context.log_extra(
+                    argument_list=argument_list,
+                    number_arguments=len(argument_list),
+                    scheduled_task_ids=scheduled_tasks.as_tuple(),
+                ),
+            )
+
         else:
             if checkpoints:
                 checkpoints.log(UploadFlow.INITIAL_PROCESSING_COMPLETE)
                 checkpoints.log(UploadFlow.NO_REPORTS_FOUND)
             log.info(
-                "Not scheduling task because there were no arguments were found on redis",
-                extra=dict(
-                    repoid=commit.repoid,
-                    commit=commit.commitid,
-                    argument_list=argument_list,
-                ),
+                "Not scheduling task because there were no arguments found on redis",
+                extra=upload_context.log_extra(),
             )
         return {"was_setup": was_setup, "was_updated": was_updated}
 
     def schedule_task(
         self,
+        db_session: Session,
         commit: Commit,
         commit_yaml: UserYaml,
         argument_list: list[dict],
         commit_report: CommitReport,
         upload_context: UploadContext,
-        db_session,
-        checkpoints=None,
+        checkpoints: CheckpointLogger | None,
     ):
         commit_yaml = commit_yaml.to_dict()
 
-        res = None
-        if (
-            commit_report.report_type is None
-            or commit_report.report_type == ReportType.COVERAGE.value
-        ):
-            res = self._schedule_coverage_processing_task(
+        if upload_context.report_type == ReportType.COVERAGE:
```

Comment:
(for my own learning) is the `match` statement a less pythonic way of doing these if / elses ? I think I would personally prefer that

Reply:
Good point, I’m not quite sure. I have only seen match/case statements in codecov, haven’t seen them anywhere in Sentry.
I don’t think python / mypy has proper exhaustiveness checking, so I don’t think there are any benefits from using a match/case statement.